std/math.d

```diff
@@ -1210,13 +1210,13 @@ private T tanImpl(T)(T x) @safe pure nothrow @nogc
             T t = tan(x);
 
             //printf("tan(%Lg) = %Lg, should be %Lg\n", cast(real) x, cast(real) t, cast(real) r);
-            assert(approxEqual(r, t));
+            assert(isClose(r, t, CommonDefaultFor!(T,T), CommonDefaultFor!(T,T)));
```

Comment:
This is a bit tricky, because the test appears inside of two `static foreach`. For most of this tests `isClose(r, t)` would be enough, but there are two tests included which compare to `0.0`. For these tests, the absolute test needs to be enabled with the fourth parameter. As these are templated tests, the floating point numbers are of different types. `CommonDefaultFor!(T,T)` calculates the best values for these tests.

Alternatively one could check if `r==0` and apply a different test in that case - this would avoid the use of the private function `CommonDefaultFor`. See the tests for `atan2` where the original tests contained such a solution. I will change that here (and at some other places too) if you think this would be better.